lib/chef/knife/ec_restore.rb

```diff
@@ -238,11 +238,23 @@ def upload_org_data(name)
           # pivotal and members of the admins group. Use the same strategy
           # above here.
           #
-          ['admins', 'billing-admins', 'public_key_read_access'].each do |group|
+          groups = ['admins', 'billing-admins']
+          if server.supports_public_key_read_access? &&
+             ::File.exist?(::File.join(chef_fs_config.local_fs.child_paths['groups'], 'public_key_read_access.json'))
+            groups.push('public_key_read_access')
+          end
+
+          groups.each do |group|
             restore_group(chef_fs_config, group, :clients => false)
           end
 
-          ['/acls/groups/billing-admins.json', '/acls/groups/public_key_read_access.json'].each do |acl|
+          acls_groups_paths = ['/acls/groups/billing-admins.json']
+          if server.supports_public_key_read_access? &&
```

Comment:
I'm wondering if we should skip the version check on restore here since, if the group is in the backup it could potentially appear in ACLs which would then fail to upload.